### ISSUE TITLE
building lapack:x64-windows failed.

### ISSUE THREAD

**Issue by forallsunday** [microsoft/vcpkg#22581]
**Host Environment**
- OS: windows 10
- Compiler: vs 2015

**To Reproduce**
vcpkg install lapack:x64-windows

**Failure logs** 
    config-x64-windows-out.log
[1/2] cmd /c "cd .. && "D:/Library/vcpkg/downloads/tools/cmake-3.21.1-windows/cmake-3.21.1-windows-i386/bin/cmake.exe" "D:/Library/vcpkg/ports/lapack" "-DCMAKE_PREFIX_PATH="D:/Library/vcpkg/packages/lapack_x64-windows"" "-DBUILD_SHARED_LIBS=ON" "-DVCPKG_CHAINLOAD_TOOLCHAIN_FILE=D:/Library/vcpkg/scripts/toolchains/windows.cmake" "-DVCPKG_TARGET_TRIPLET=x64-windows" "-DVCPKG_SET_CHARSET_FLAG=ON" "-DVCPKG_PLATFORM_TOOLSET=v140" "-DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_SYSTEM_PACKAGE_REGISTRY=ON" "-DCMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_SKIP=TRUE" "-DCMAKE_VERBOSE_MAKEFILE=ON" "-DVCPKG_APPLOCAL_DEPS=OFF" "-DCMAKE_TOOLCHAIN_FILE=D:/Library/vcpkg/scripts/buildsystems/vcpkg.cmake" "-DCMAKE_ERROR_ON_ABSOLUTE_INSTALL_DESTINATION=ON" "-DVCPKG_CXX_FLAGS=" "-DVCPKG_CXX_FLAGS_RELEASE=" "-DVCPKG_CXX_FLAGS_DEBUG=" "-DVCPKG_C_FLAGS=" "-DVCPKG_C_FLAGS_RELEASE=" "-DVCPKG_C_FLAGS_DEBUG=" "-DVCPKG_CRT_LINKAGE=dynamic" "-DVCPKG_LINKER_FLAGS=" "-DVCPKG_LINKER_FLAGS_RELEASE=" "-DVCPKG_LINKER_FLAGS_DEBUG=" "-DVCPKG_TARGET_ARCHITECTURE=x64" "-DCMAKE_INSTALL_LIBDIR:STRING=lib" "-DCMAKE_INSTALL_BINDIR:STRING=bin" "-D_VCPKG_ROOT_DIR=D:/Library/vcpkg" "-DZ_VCPKG_ROOT_DIR=D:/Library/vcpkg" "-D_VCPKG_INSTALLED_DIR=D:/Library/vcpkg/installed" "-DVCPKG_MANIFEST_INSTALL=OFF" "-Ax64" "-G" "Visual Studio 14 2015" "-DCMAKE_BUILD_TYPE=Release" "-DCMAKE_INSTALL_PREFIX=D:/Library/vcpkg/packages/lapack_x64-windows""
FAILED: ../CMakeCache.txt 
cmd /c "cd .. && "D:/Library/vcpkg/downloads/tools/cmake-3.21.1-windows/cmake-3.21.1-windows-i386/bin/cmake.exe" "D:/Library/vcpkg/ports/lapack" "-DCMAKE_PREFIX_PATH="D:/Library/vcpkg/packages/lapack_x64-windows"" "-DBUILD_SHARED_LIBS=ON" "-DVCPKG_CHAINLOAD_TOOLCHAIN_FILE=D:/Library/vcpkg/scripts/toolchains/windows.cmake" "-DVCPKG_TARGET_TRIPLET=x64-windows" "-DVCPKG_SET_CHARSET_FLAG=ON" "-DVCPKG_PLATFORM_TOOLSET=v140" "-DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_SYSTEM_PACKAGE_REGISTRY=ON" "-DCMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_SKIP=TRUE" "-DCMAKE_VERBOSE_MAKEFILE=ON" "-DVCPKG_APPLOCAL_DEPS=OFF" "-DCMAKE_TOOLCHAIN_FILE=D:/Library/vcpkg/scripts/buildsystems/vcpkg.cmake" "-DCMAKE_ERROR_ON_ABSOLUTE_INSTALL_DESTINATION=ON" "-DVCPKG_CXX_FLAGS=" "-DVCPKG_CXX_FLAGS_RELEASE=" "-DVCPKG_CXX_FLAGS_DEBUG=" "-DVCPKG_C_FLAGS=" "-DVCPKG_C_FLAGS_RELEASE=" "-DVCPKG_C_FLAGS_DEBUG=" "-DVCPKG_CRT_LINKAGE=dynamic" "-DVCPKG_LINKER_FLAGS=" "-DVCPKG_LINKER_FLAGS_RELEASE=" "-DVCPKG_LINKER_FLAGS_DEBUG=" "-DVCPKG_TARGET_ARCHITECTURE=x64" "-DCMAKE_INSTALL_LIBDIR:STRING=lib" "-DCMAKE_INSTALL_BINDIR:STRING=bin" "-D_VCPKG_ROOT_DIR=D:/Library/vcpkg" "-DZ_VCPKG_ROOT_DIR=D:/Library/vcpkg" "-D_VCPKG_INSTALLED_DIR=D:/Library/vcpkg/installed" "-DVCPKG_MANIFEST_INSTALL=OFF" "-Ax64" "-G" "Visual Studio 14 2015" "-DCMAKE_BUILD_TYPE=Release" "-DCMAKE_INSTALL_PREFIX=D:/Library/vcpkg/packages/lapack_x64-windows""
-- Selecting Windows SDK version 10.0.14393.0 to target Windows 10.0.19042.
-- The C compiler identification is unknown
-- The CXX compiler identification is unknown
CMake Error at CMakeLists.txt:3 (project):
  No CMAKE_C_COMPILER could be found.

CMake Error at CMakeLists.txt:3 (project):
  No CMAKE_CXX_COMPILER could be found.

-- Configuring incomplete, errors occurred!
See also "D:/Library/vcpkg/buildtrees/lapack/x64-windows-rel/CMakeFiles/CMakeOutput.log".
See also "D:/Library/vcpkg/buildtrees/lapack/x64-windows-rel/CMakeFiles/CMakeError.log".
[2/2] cmd /c "cd ../../x64-windows-dbg && "D:/Library/vcpkg/downloads/tools/cmake-3.21.1-windows/cmake-3.21.1-windows-i386/bin/cmake.exe" "D:/Library/vcpkg/ports/lapack" "-DCMAKE_PREFIX_PATH="D:/Library/vcpkg/packages/lapack_x64-windows"" "-DBUILD_SHARED_LIBS=ON" "-DVCPKG_CHAINLOAD_TOOLCHAIN_FILE=D:/Library/vcpkg/scripts/toolchains/windows.cmake" "-DVCPKG_TARGET_TRIPLET=x64-windows" "-DVCPKG_SET_CHARSET_FLAG=ON" "-DVCPKG_PLATFORM_TOOLSET=v140" "-DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_SYSTEM_PACKAGE_REGISTRY=ON" "-DCMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_SKIP=TRUE" "-DCMAKE_VERBOSE_MAKEFILE=ON" "-DVCPKG_APPLOCAL_DEPS=OFF" "-DCMAKE_TOOLCHAIN_FILE=D:/Library/vcpkg/scripts/buildsystems/vcpkg.cmake" "-DCMAKE_ERROR_ON_ABSOLUTE_INSTALL_DESTINATION=ON" "-DVCPKG_CXX_FLAGS=" "-DVCPKG_CXX_FLAGS_RELEASE=" "-DVCPKG_CXX_FLAGS_DEBUG=" "-DVCPKG_C_FLAGS=" "-DVCPKG_C_FLAGS_RELEASE=" "-DVCPKG_C_FLAGS_DEBUG=" "-DVCPKG_CRT_LINKAGE=dynamic" "-DVCPKG_LINKER_FLAGS=" "-DVCPKG_LINKER_FLAGS_RELEASE=" "-DVCPKG_LINKER_FLAGS_DEBUG=" "-DVCPKG_TARGET_ARCHITECTURE=x64" "-DCMAKE_INSTALL_LIBDIR:STRING=lib" "-DCMAKE_INSTALL_BINDIR:STRING=bin" "-D_VCPKG_ROOT_DIR=D:/Library/vcpkg" "-DZ_VCPKG_ROOT_DIR=D:/Library/vcpkg" "-D_VCPKG_INSTALLED_DIR=D:/Library/vcpkg/installed" "-DVCPKG_MANIFEST_INSTALL=OFF" "-Ax64" "-G" "Visual Studio 14 2015" "-DCMAKE_BUILD_TYPE=Debug" "-DCMAKE_INSTALL_PREFIX=D:/Library/vcpkg/packages/lapack_x64-windows/debug""
FAILED: ../../x64-windows-dbg/CMakeCache.txt 
cmd /c "cd ../../x64-windows-dbg && "D:/Library/vcpkg/downloads/tools/cmake-3.21.1-windows/cmake-3.21.1-windows-i386/bin/cmake.exe" "D:/Library/vcpkg/ports/lapack" "-DCMAKE_PREFIX_PATH="D:/Library/vcpkg/packages/lapack_x64-windows"" "-DBUILD_SHARED_LIBS=ON" "-DVCPKG_CHAINLOAD_TOOLCHAIN_FILE=D:/Library/vcpkg/scripts/toolchains/windows.cmake" "-DVCPKG_TARGET_TRIPLET=x64-windows" "-DVCPKG_SET_CHARSET_FLAG=ON" "-DVCPKG_PLATFORM_TOOLSET=v140" "-DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_SYSTEM_PACKAGE_REGISTRY=ON" "-DCMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_SKIP=TRUE" "-DCMAKE_VERBOSE_MAKEFILE=ON" "-DVCPKG_APPLOCAL_DEPS=OFF" "-DCMAKE_TOOLCHAIN_FILE=D:/Library/vcpkg/scripts/buildsystems/vcpkg.cmake" "-DCMAKE_ERROR_ON_ABSOLUTE_INSTALL_DESTINATION=ON" "-DVCPKG_CXX_FLAGS=" "-DVCPKG_CXX_FLAGS_RELEASE=" "-DVCPKG_CXX_FLAGS_DEBUG=" "-DVCPKG_C_FLAGS=" "-DVCPKG_C_FLAGS_RELEASE=" "-DVCPKG_C_FLAGS_DEBUG=" "-DVCPKG_CRT_LINKAGE=dynamic" "-DVCPKG_LINKER_FLAGS=" "-DVCPKG_LINKER_FLAGS_RELEASE=" "-DVCPKG_LINKER_FLAGS_DEBUG=" "-DVCPKG_TARGET_ARCHITECTURE=x64" "-DCMAKE_INSTALL_LIBDIR:STRING=lib" "-DCMAKE_INSTALL_BINDIR:STRING=bin" "-D_VCPKG_ROOT_DIR=D:/Library/vcpkg" "-DZ_VCPKG_ROOT_DIR=D:/Library/vcpkg" "-D_VCPKG_INSTALLED_DIR=D:/Library/vcpkg/installed" "-DVCPKG_MANIFEST_INSTALL=OFF" "-Ax64" "-G" "Visual Studio 14 2015" "-DCMAKE_BUILD_TYPE=Debug" "-DCMAKE_INSTALL_PREFIX=D:/Library/vcpkg/packages/lapack_x64-windows/debug""
-- Selecting Windows SDK version 10.0.14393.0 to target Windows 10.0.19042.
-- The C compiler identification is unknown
-- The CXX compiler identification is unknown
CMake Error at CMakeLists.txt:3 (project):
  No CMAKE_C_COMPILER could be found.

CMake Error at CMakeLists.txt:3 (project):
  No CMAKE_CXX_COMPILER could be found.

-- Configuring incomplete, errors occurred!
See also "D:/Library/vcpkg/buildtrees/lapack/x64-windows-dbg/CMakeFiles/CMakeOutput.log".
See also "D:/Library/vcpkg/buildtrees/lapack/x64-windows-dbg/CMakeFiles/CMakeError.log".
ninja: build stopped: subcommand failed.

**Additional context**
my cpu is amd.

**Comment by forallsunday** [comment]
Now I can't build any project for cmake can't find compiler of vs2015 x64.
```
-- The C compiler identification is unknown
-- The CXX compiler identification is unknown
CMake Error at CMakeLists.txt:3 (project):
No CMAKE_C_COMPILER could be found.

CMake Error at CMakeLists.txt:3 (project):
No CMAKE_CXX_COMPILER could be found.
```
When I change the name of 'vcpkg' dir, things worked fine, seems cmake reading some configurations from vcpkg.

**Comment by forallsunday** [comment]
The purpose to build lapack is to build ceres with suitesparse. I change the dependcy of lapack from lapack-referece to clapack, and I finally built the ceres[suitesparse] successfully.

**Comment by LilyWangLL** [comment]
> The purpose to build lapack is to build ceres with suitesparse. I change the dependcy of lapack from lapack-referece to clapack, and I finally built the ceres[suitesparse] successfully.

This issue should be fixed by #21479.

**Comment by LilyWangLL** [comment]
`lapack-reference` and `lapack` could install succeed on the latest vcpkg.
```
Installing 4/5 lapack-reference[blas-select,core,noblas]:x64-windows@3.11.0#6...
Building lapack-reference[blas-select,core,noblas]:x64-windows@3.11.0#6...
-- Downloading https://github.com/Reference-LAPACK/lapack/archive/v3.11.0.tar.gz -> Reference-LAPACK-lapack-v3.11.0.tar.gz...
-- Extracting source F:/0624/downloads/Reference-LAPACK-lapack-v3.11.0.tar.gz
-- Applying patch cmake-config.patch
-- Applying patch lapacke.patch
-- Applying patch fix_prefix.patch
-- Using source at F:/0624/buildtrees/lapack-reference/src/v3.11.0-6ae738f586.clean
-- The Fortran compiler identification is unknown
-- No Fortran compiler found on the PATH. Using MinGW gfortran!
-- Using cached msys2-mingw-w64-x86_64-gcc-fortran-14.1.0-3-any.pkg.tar.zst.
-- Using cached msys2-msys2-runtime-3.5.3-3-x86_64.pkg.tar.zst.
-- Using cached msys2-mingw-w64-x86_64-gcc-14.1.0-3-any.pkg.tar.zst.
-- Using cached msys2-mingw-w64-x86_64-gcc-libgfortran-14.1.0-3-any.pkg.tar.zst.
-- Using cached msys2-mingw-w64-x86_64-binutils-2.42-2-any.pkg.tar.zst.
-- Using cached msys2-mingw-w64-x86_64-crt-git-12.0.0.r32.gf977e1c38-1-any.pkg.tar.zst.
-- Using cached msys2-mingw-w64-x86_64-gcc-libs-14.1.0-3-any.pkg.tar.zst.
-- Using cached msys2-mingw-w64-x86_64-gmp-6.3.0-2-any.pkg.tar.zst.
-- Using cached msys2-mingw-w64-x86_64-headers-git-12.0.0.r32.gf977e1c38-1-any.pkg.tar.zst.
-- Using cached msys2-mingw-w64-x86_64-isl-0.26-1-any.pkg.tar.zst.
-- Using cached msys2-mingw-w64-x86_64-libiconv-1.17-4-any.pkg.tar.zst.
-- Using cached msys2-mingw-w64-x86_64-mpc-1.3.1-2-any.pkg.tar.zst.
-- Using cached msys2-mingw-w64-x86_64-mpfr-4.2.1-2-any.pkg.tar.zst.
-- Using cached msys2-mingw-w64-x86_64-windows-default-manifest-6.4-4-any.pkg.tar.zst.
-- Using cached msys2-mingw-w64-x86_64-winpthreads-git-12.0.0.r32.gf977e1c38-1-any.pkg.tar.zst.
-- Using cached msys2-mingw-w64-x86_64-zlib-1.3.1-1-any.pkg.tar.zst.
-- Using cached msys2-mingw-w64-x86_64-zstd-1.5.6-2-any.pkg.tar.zst.
-- Using cached msys2-mingw-w64-x86_64-gettext-runtime-0.22.5-2-any.pkg.tar.zst.
-- Using cached msys2-mingw-w64-x86_64-headers-git-12.0.0.r32.gf977e1c38-1-any.pkg.tar.zst.
-- Using cached msys2-mingw-w64-x86_64-libwinpthread-git-12.0.0.r32.gf977e1c38-1-any.pkg.tar.zst.
-- Using cached msys2-mingw-w64-x86_64-libwinpthread-git-12.0.0.r32.gf977e1c38-1-any.pkg.tar.zst.
-- Using msys root at F:/0624/downloads/tools/msys2/259fff2112122ad2
-- Found external ninja('1.11.0').
-- Configuring x64-windows
-- Building x64-windows-dbg
-- Building x64-windows-rel
-- Fixing pkgconfig file: F:/0624/packages/lapack-reference_x64-windows/lib/pkgconfig/lapack.pc
-- Using cached msys2-mingw-w64-x86_64-pkgconf-1~2.2.0-1-any.pkg.tar.zst.
-- Using cached msys2-msys2-runtime-3.5.3-3-x86_64.pkg.tar.zst.
-- Using msys root at F:/0624/downloads/tools/msys2/1e74ca60daa10104
-- Fixing pkgconfig file: F:/0624/packages/lapack-reference_x64-windows/debug/lib/pkgconfig/lapack.pc
-- Installing: F:/0624/packages/lapack-reference_x64-windows/share/lapack-reference/usage
-- Installing: F:/0624/packages/lapack-reference_x64-windows/share/lapack-reference/copyright
-- Performing post-build validation
Elapsed time to handle lapack-reference:x64-windows: 2.5 min
lapack-reference:x64-windows package ABI: ff441f7403aae0d90a44c7afb94491ba86a14eab7aa612a3a4e6f41bdb7bdc69
Installing 5/5 lapack:x64-windows@2023-06-10#2...
Building lapack:x64-windows@2023-06-10#2...
-- Skipping post-build validation due to VCPKG_POLICY_EMPTY_PACKAGE
Elapsed time to handle lapack:x64-windows: 90.7 ms
lapack:x64-windows package ABI: 3c6d866748126abde365d26f2158ab38fefb5a4d922846a4b967dee7175c84f0
Total install time: 6.4 min
lapack provides pkg-config modules:

  # Implementation of LAPACK
  lapack
```
